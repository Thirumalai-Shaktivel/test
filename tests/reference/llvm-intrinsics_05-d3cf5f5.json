{
    "basename": "llvm-intrinsics_05-d3cf5f5",
    "cmd": "lfortran --show-llvm {infile} -o {outfile}",
    "infile": "tests/../integration_tests/intrinsics_05.f90",
    "infile_hash": "735edff1f9942ce107c1b84f9ddcf00e648e553327f54dacf7b8238c",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm-intrinsics_05-d3cf5f5.stdout",
    "stdout_hash": "4a803c2c3159026a13987073724787ac34ccbe785226578175f90376",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}