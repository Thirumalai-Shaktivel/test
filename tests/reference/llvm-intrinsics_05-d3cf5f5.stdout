; ModuleID = 'LFortran'
source_filename = "LFortran"

@0 = private unnamed_addr constant [4 x i8] c"%f\0A\00", align 1
@1 = private unnamed_addr constant [4 x i8] c"%f\0A\00", align 1
@2 = private unnamed_addr constant [4 x i8] c"%f\0A\00", align 1

define float @__module_lfortran_intrinsic_math_scosh(float* %x) {
.entry:
  %r = alloca float, align 4
  %0 = load float, float* %x, align 4
  %1 = call float @_lfortran_scosh(float %0)
  store float %1, float* %r, align 4
  %2 = load float, float* %r, align 4
  ret float %2
}

declare float @_lfortran_scosh(float)

define float @__module_lfortran_intrinsic_math_ssinh(float* %x) {
.entry:
  %r = alloca float, align 4
  %0 = load float, float* %x, align 4
  %1 = call float @_lfortran_ssinh(float %0)
  store float %1, float* %r, align 4
  %2 = load float, float* %r, align 4
  ret float %2
}

declare float @_lfortran_ssinh(float)

define float @__module_lfortran_intrinsic_math_stanh(float* %x) {
.entry:
  %r = alloca float, align 4
  %0 = load float, float* %x, align 4
  %1 = call float @_lfortran_stanh(float %0)
  store float %1, float* %r, align 4
  %2 = load float, float* %r, align 4
  ret float %2
}

declare float @_lfortran_stanh(float)

define i32 @main() {
.entry:
  %x = alloca float, align 4
  %0 = alloca float, align 4
  store float 1.000000e+00, float* %0, align 4
  %1 = call float @__module_lfortran_intrinsic_math_ssinh(float* %0)
  store float %1, float* %x, align 4
  %2 = load float, float* %x, align 4
  %3 = fpext float %2 to double
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @0, i32 0, i32 0), double %3)
  %4 = alloca float, align 4
  store float 1.000000e+00, float* %4, align 4
  %5 = call float @__module_lfortran_intrinsic_math_scosh(float* %4)
  store float %5, float* %x, align 4
  %6 = load float, float* %x, align 4
  %7 = fpext float %6 to double
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @1, i32 0, i32 0), double %7)
  %8 = alloca float, align 4
  store float 1.000000e+00, float* %8, align 4
  %9 = call float @__module_lfortran_intrinsic_math_stanh(float* %8)
  store float %9, float* %x, align 4
  %10 = load float, float* %x, align 4
  %11 = fpext float %10 to double
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @2, i32 0, i32 0), double %11)
  ret i32 0
}

declare void @_lfortran_printf(i8*, ...)

